### PR TITLE
DOCS-4346 Profile types page

### PR DESCRIPTION
**Pull request by kayayarai**
### What does this PR do?
Splits profile types info into its own page, and copies language version support info (esp Python) onto the same place. Adds tracing library minimum version for Code hotspots and Endpoints profiling features.

### Motivation
request from PM, DOCS-4346

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/profiler/_index.md [hunk 1]
```diff
@@ -59,7 +59,15 @@ The [Getting Started with Profiler][1] guide takes a sample service with a perfo
 
 ## Explore Datadog profiler
 
-After you've configured your application to send profiles to Datadog, start getting insights into your code performance. By default, profiles are retained for 7 days, and metrics generated from profile data are retained for 1 month.
+After you've configured your application to send profiles to Datadog, start getting insights into your code performance.  
+
+By default, profiles are retained for 7 days, and metrics generated from profile data are retained for 1 month.
```
Comment (cswatt) [inline content/en/profiler/_index.md:64]:
```suggestion
By default, profiles are retained for seven days, and metrics generated from profile data are retained for one month.
```

File: content/en/profiler/enabling/go.md [hunk 2]
```diff
@@ -22,7 +22,9 @@ The profiler is shipped within Datadog tracing libraries. If you are already usi
 
 ## Requirements
 
-The Datadog Profiler requires Go 1.12+.
+The Datadog Profiler requires Go 1.12+. For Endpoints profiling.
```
Comment (cswatt) [inline content/en/profiler/enabling/go.md:25]:
```suggestion
The Datadog Profiler requires Go 1.12+ for Endpoints profiling.
```

File: content/en/profiler/enabling/go.md [hunk 3]
```diff
@@ -22,7 +22,9 @@ The profiler is shipped within Datadog tracing libraries. If you are already usi
...
+
+For Code Hotspots and Endpoints profiling, use Go version 1.18 or greater and `dd-trace-go` version 1.37.0 or greater.
```
Comment (cswatt) [inline content/en/profiler/enabling/go.md:27]: from here, it looks like Endpoints requires Go 1.12+ and dd-trace-go 1.37.0+, and Code Hotspots requires Go 1.18+ and dd-trace-go 1.37.0+ ..... is there a way to combine the information in lines 25-27 for more clarity?

File: content/en/profiler/enabling/java.md [hunk 4]
```diff
@@ -28,10 +28,15 @@ The Datadog Profiler requires [JDK Flight Recorder][2]. The Datadog Profiler lib
 
 Because non-LTS JDK versions may not contain stability and performance fixes related to the Datadog Profiler library, use versions 8, 11, and 17 of the Long Term Support JDK.
 
+Additional requirements profiling [code hotspots][11] feature:
```
Comment (cswatt) [inline content/en/profiler/enabling/java.md:31]:
```suggestion
Additional requirements for profiling [Code Hotspots][11]:
```

seems like we're capitalizing Code Hotspots

File: content/en/profiler/profile_types.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+Block
+: The time functions have been waiting on mutexes and channel operations during the profiling period (default: 60s). Sleep, GC, Network, and Syscall operations are not captured by this profile. Blocking operations are only captured after they become unblocked, so this profile cannot be used to debug applications that appear to be stuck. For mutex contentions, the stack traces in this profile point to blocked `Lock()` operations. This tells you where your program is getting blocked, while the mutex profile tells you what part of your program is causing the contention. See our [Block Profiling in Go][1] research for more in-depth information. See also the note about how this measure changes in version `1.33.0` in [Delta profiles](#delta-profiles). **Note:** The block profiler can cause noticeable overhead for production workloads. If enabling it in production, prefer high rates (such as `100000000`, which is 100 milliseconds) and look for signs of increased latency or CPU utilization.
```
Comment (cswatt) [inline content/en/profiler/profile_types.md:123]:
```suggestion
: The time functions have been waiting on mutexes and channel operations during the profiling period (default: 60s). Sleep, GC, Network, and Syscall operations are not captured by this profile. Blocking operations are only captured after they become unblocked, so this profile cannot be used to debug applications that appear to be stuck. For mutex contentions, the stack traces in this profile point to blocked `Lock()` operations. This tells you where your program is getting blocked, while the mutex profile tells you what part of your program is causing the contention. See Datadog's [Block Profiling in Go][1] research for more in-depth information. See also the note about how this measure changes in version `1.33.0` in [Delta profiles](#delta-profiles). **Note**: The block profiler can cause noticeable overhead for production workloads. If enabling it in production, prefer high rates (such as `100000000`, which is 100 milliseconds) and look for signs of increased latency or CPU utilization.
```

File: content/en/profiler/profile_types.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+
+Goroutines
+: A snapshot of the number of goroutines currently executing the same functions (both on-CPU and waiting off-CPU). An increasing number of goroutines between snapshots can indicate that the program is leaking goroutines. In most healthy applications this profile is dominated by worker pools and the number of goroutines they use. Applications that are extremely latency-sensitive and use a large number of goroutines (> 10.000) should be aware that enabling this profile requires stop-the-world pauses. The pauses occur only once every profiling period (default 60s) and normally last for around `1µsec` per goroutine. Typical applications with a p99 latency SLO of around `100ms` can generally ignore this warning. See our [Goroutine Profiling in Go][2] research for more in-depth information.
```
Comment (cswatt) [inline content/en/profiler/profile_types.md:126]:
```suggestion
: A snapshot of the number of goroutines currently executing the same functions (both on-CPU and waiting off-CPU). An increasing number of goroutines between snapshots can indicate that the program is leaking goroutines. In most healthy applications this profile is dominated by worker pools and the number of goroutines they use. Applications that are extremely latency-sensitive and use a large number of goroutines (> 10.000) should be aware that enabling this profile requires stop-the-world pauses. The pauses occur only once every profiling period (default 60s) and normally last for around `1µsec` per goroutine. Typical applications with a p99 latency SLO of around `100ms` can generally ignore this warning. See Datadog's [Goroutine Profiling in Go][2] research for more in-depth information.
```

**Review by cswatt — CHANGES_REQUESTED** [review]

File: content/en/profiler/enabling/go.md (hunk 2 above)
Comment (kayayarai) [reply content/en/profiler/enabling/go.md:25]: Ah, this is a stray paste. Endpoints and CH both require 1.18+

Comment (kayayarai) [inline content/en/profiler/enabling/go.md:25]:
```suggestion
The Datadog Profiler requires Go 1.12+.
```

File: content/en/profiler/_index.md [hunk 7]
```diff
@@ -59,7 +59,15 @@ The [Getting Started with Profiler][1] guide takes a sample service with a perfo
 
 ## Explore Datadog profiler
 
-After you've configured your application to send profiles to Datadog, start getting insights into your code performance. By default, profiles are retained for 7 days, and metrics generated from profile data are retained for 1 month.
+After you've configured your application to send profiles to Datadog, start getting insights into your code performance.
```
Comment (kayayarai) [inline content/en/profiler/_index.md:62]:
```suggestion
After you configure your application to send profiles to Datadog, start getting insights into your code performance.  
```

**Review by cswatt — APPROVED** [review]

**Pull request merged**